lib/logstash-logger/device/balancer.rb

```diff
@@ -0,0 +1,40 @@
+module LogStashLogger
+  module Device
+    class Balancer < Base
+      attr_reader :devices
+
+      def initialize(opts)
+        @io = self
+        @devices = create_devices(opts[:outputs])
+        self.class.delegate_to_all(:close, :flush)
+        self.class.delegate_to_one(:write)
+      end
+
+      private
+
+      def create_devices(opts)
+        opts.map { |o| Device.new(o) }
+      end
+
+      def self.delegate_to_all(*methods)
+        methods.each do |m|
+          define_method(m) do |*args|
+            @devices.each { |device| device.send(m, *args) }
+          end
+        end
+      end
+
+      def self.delegate_to_one(*methods)
+        methods.each do |m|
+          define_method(m) do |*args|
+            select_device(*args).send(m, *args)
+          end
+        end
+      end
+
+      def select_device(args)
+        devices[Zlib::crc32(args) % devices.count]
```

Comment:
What's the advantage of this over `devices.sample`? It seems to me like this way could potentially get unbalanced, sending more writes to some devices than others.